apis/discovery/v1alpha1/foreigncluster_types.go

```diff
@@ -104,6 +109,18 @@ type ClusterIdentity struct {
 	ClusterName string `json:"clusterName"`
 }
 
+// UniqueName returns a user-friendly name that is unique *in the context of the current cluster*.
+// It depends on ClusterName, so the same cluster may have different UniqueNames in different clusters.
+//
+// Use it when reflecting resources on a remote cluster (see issue #966).
+func (i *ClusterIdentity) UniqueName() string {
```

Comment:
I think it should be defined elsewhere not in the API pkg.

Reply:
This issue is still un-addressed. Do you think there is a better package for these?